### PR TITLE
Clarify LocationConstraint parameter

### LINKED ISSUE #988: Document bucket-specific location handling
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I don't know how to choose between different placement policies.

## Describe the solution you'd like
Specify how we treat `LocationConstraint` in S3 API documentation.

### PR DESCRIPTION
**Pull request by smallhive**
Close #988.

### CODE REVIEW

File: docs/aws_s3_compat.md [hunk 1]
```diff
@@ -170,6 +170,12 @@ See also `GetObject` and other method parameters.
 | 🟢 | ListBuckets          |           |
 | 🔵 | PutPublicAccessBlock |           |
 
+* The `CreateBucket` method allows you to select a placement policy using the `LocationConstraint` parameter in the AWS CLI. The policy name should be passed as a value.
```
Comment (roman-khimov) [inline docs/aws_s3_compat.md:173]: `s/The//`, some "Extensions" section would be nice to have.

File: docs/aws_s3_compat.md [hunk 2]
```diff
@@ -170,6 +170,12 @@ See also `GetObject` and other method parameters.
...
+  * Policies mapping can be defined via:
+    * [Authmate](./authmate.md#containers-policy) during buket creation. These policies are available only for generated credentials.
```
Comment (roman-khimov) [inline docs/aws_s3_compat.md:175]: s/buket/bucket/

File: api/layer/layer.go [hunk 3]
```diff
@@ -569,6 +569,10 @@ func (n *layer) deleteObject(ctx context.Context, bkt *data.BucketInfo, settings
 		}
 
 		if obj.DeleteMarkVersion, obj.Error = n.removeOldVersion(ctx, bkt, nodeVersion, obj); obj.Error != nil {
+			if strings.Contains(obj.Error.Error(), "2050 message = object is locked") {
```
Comment (roman-khimov) [inline api/layer/layer.go:572]: How is it related to #988?

File: docs/aws_s3_compat.md (hunk 2 above)
Comment (smallhive) [reply docs/aws_s3_compat.md]: Fixed

File: api/layer/layer.go (hunk 3 above)
Comment (smallhive) [reply api/layer/layer.go]: My fault. It is not related. Wrong branch

File: docs/aws_s3_compat.md (hunk 1 above)
Comment (smallhive) [reply docs/aws_s3_compat.md]: Removed _The_

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**